public/app/plugins/datasource/prometheus/querybuilder/components/MetricSelect.tsx

```diff
@@ -12,12 +16,39 @@ export interface Props {
 }
 
 export function MetricSelect({ query, onChange, onGetMetrics }: Props) {
-  const styles = getStyles();
+  const styles = useStyles2(getStyles);
   const [state, setState] = useState<{
     metrics?: Array<SelectableValue<any>>;
     isLoading?: boolean;
   }>({});
 
+  const customFilterOption = useCallback((option: SelectableValue<any>, searchQuery: string) => {
+    const label = option.label ?? option.value;
+    if (!label) {
+      return false;
+    }
+    const searchWords = searchQuery.split(splitSeparator);
+    return searchWords.reduce((acc, cur) => acc && label.toLowerCase().includes(cur.toLowerCase()), true);
+  }, []);
+
+  const formatOptionLabel = useCallback(
+    (option: SelectableValue<any>, meta: FormatOptionLabelMeta<any>) => {
+      // For newly created custom value we don't want to add highlight
+      if (option['__isNew__']) {
+        return option.label;
+      }
+
+      return (
+        <Highlighter
+          searchWords={meta.inputValue.split(splitSeparator)}
+          textToHighlight={option.label ?? ''}
+          highlightClassName={styles.hightlight}
+        />
+      );
+    },
+    [styles.hightlight]
+  );
```

Comment:
would be interesting topic for frontend weekly or #frontend and #query-builders,  feels like this feature (Multi word search and highlight) could great in other query editors .

Reply:
I've added it to frontend weekly next week's agenda: https://docs.google.com/document/d/1LmS2esCEGI1eacQMgg-0ZyRjg84kRFw6Z1XIqkCeOIc/edit#heading=h.6bso5590olwu